lib/active_record/virtual_attributes/virtual_delegates.rb

```diff
@@ -247,6 +251,7 @@ def self.select_from_alias(to_ref, col, to_model_col_name, src_model_id)
         to_table    = select_from_alias_table(to_ref.klass, src_model_id.relation)
         to_model_id = to_ref.klass.arel_attribute(to_model_col_name, to_table)
         to_column   = to_ref.klass.arel_attribute(col, to_table)
+        raise "invalid virtual delegate referencing #{to_ref.name}.#{col}" unless to_column
```

Comment:
Ah, is this why we can remove the `to_ref.klass.arel_attribute(col)` check above?

Reply:
yea, otherwise we end up with a `select(nil)` below and good luck debugging that one.

Reply:
No, that line was useless in my tests. still end up going to the database - and the sql is bad.
So we'll need tests around the sql for virtual attributes now. I've been adding tests when ever I modify a virtual attribute/delegate - it is easy to botch these up.